<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="go1" doc="Robot type: [a1, aliengo, go1, laikago]"/>
    <arg name="use_js" default="false"/>
    <arg name="cheater" default="false"/>
      <!--$(env ROBOT_TYPE)  -->
    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
"/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
"/>
    <param name="ckpt_path" value="/home/guoyi/Desktop/unitree_sim2real/src/legged_control/rl_policy_ros/ckpt/model_60000000.pt"/>
<!--    <node pkg="rl_policy_ros" type="rl_server_node.py" name="rl_server_node" output="screen"/>-->

    <!-- 对应 Python 里 rospy.get_param("~ckpt_path") -->


    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen"/>
    <node pkg="joy" type="joy_node" name="joy_node"
          output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen"/>
    <node pkg="legged_controllers" type="joystick_vel_node" name="joystick_vel_node"
          output="screen"/>
   <group if="$(arg use_js)">
            <include file="$(find legged_controllers)/launch/joy_teleop.launch"/>
    </group>
</launch>
